File: frontend/bff/src/application/files/dtos/file.dto.ts
import type { FileKind } from '../../../cross-cutting/enums/file-kind.js';
import type { FileStatus } from '../../../cross-cutting/enums/file-status.js';

export interface FileDto {
  id: string;
  kind: FileKind;
  status: FileStatus;
  fileName: string;
  contentType: string;
  sizeBytes: number;
  organizationId: string;
  uploadedBy: string;
  uploadedAt: string;
  finishedAt: string | null;
  totalLines: number;
  processedLines: number;
  succeededLines: number;
  failedLines: number;
  /** Falha do arquivo inteiro (ex.: não pôde ser lido). */
  error: string | null;
  /** s3://bucket/chave */
  storageUrl: string;
}

File: backend/storage-service/src/Fix.Storage.Application/UseCases/Files/Dtos/FileDto.cs
using Fix.Storage.Domain.AggregateRoots.Files;

namespace Fix.Storage.Application.Files.Dtos;

public sealed record FileDto(
    Guid Id,
    FileKind Kind,
    FileStatus Status,
    string FileName,
    string ContentType,
    long SizeBytes,
    Guid OrganizationId,
    Guid UploadedBy,
    DateTimeOffset UploadedAt,
    DateTimeOffset? FinishedAt,
    int TotalLines,
    int ProcessedLines,
    int SucceededLines,
    int FailedLines,
    // Falha do arquivo inteiro (ex.: não pôde ser lido).
    string? Error,
    string StorageUrl);
